server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexStateService.java

```diff
@@ -494,4 +536,16 @@ private static int getTotalShardCount(ClusterState state, Index index) {
         return indexMetaData.getNumberOfShards() * (1 + indexMetaData.getNumberOfReplicas());
     }
 
+    /**
+     * @return Generates a {@link ClusterBlock} that blocks read and write operations on soon-to-be-closed indices. The
+     * cluster block is generated with the id value equals to {@link #INDEX_CLOSED_BLOCK_ID} and a unique UUID.
+     */
+    public static ClusterBlock createIndexClosedBlock() {
```

Comment:
Maybe call this `createIndexClosingBlock`? It's a block that's used for the time where the index is in the process of closing, not already closed